### PR TITLE
Fix Password Reset w/ Multiple Accounts

### PR DESCRIPTION
**Pull request by JVickery-TBS**
fix(user): pass reset multiple;

- Fix password reset multiple user accounts.

### CODE REVIEW

File: ckan/views/user.py [hunk 1]
```diff
@@ -656,20 +656,20 @@ def post(self):
...
         else:
             # Search by user name
             # (this is helpful as an option for a user who has multiple
             # accounts with the same email address and they want to be
             # specific)
             try:
-                logic.get_action(u'user_show')(context, {u'id': id})
-                user_objs.append(context[u'user_obj'])
+                # do `user_show` to check for user existing
+                user_dict = logic.get_action(u'user_show')(context,
```
Comment (wardi) [inline ckan/views/user.py:670]: we just called user_list, what does calling user_show on each user give us?

Comment (JVickery-TBS) [reply ckan/views/user.py:670]: The `user_list` call was in the above condition, checking for emails with `if '@' in id`.

This part is for usernames. Does calling `model.User.get()` with a bogus username raise an error, or just come back empty? If it just comes back empty, then I will do:

 ```
user_obj = model.User.get()
if user_obj:
    user_objs.append(user_ibj)
 ```

Comment (wardi) [reply ckan/views/user.py:670]: right, `else:` statements are important :-)

Yes you will get back `None` from User.get(x) when x isn't found.

Comment (JVickery-TBS) [reply ckan/views/user.py]: @wardi okay cool, I have super simplified it now

**Review by wardi — APPROVED** [review]

**Pull request merged**